src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1630,6 +1633,112 @@ export class ResultsViewPageStore {
+                        return profile.molecularAlterationType === AlterationTypeConstants.STRUCTURAL_VARIANT;
+                    }).map(profile => {
+                        return profile.molecularProfileId;
+                    });
+                    params.molecularProfileIds = molecularProfiles;
+                }
+            }
+            return await client.fetchStructuralVariantsUsingPOST({'structuralVariantFilter':<StructuralVariantFilter> params});
+        }
+    });
+
+    /**
+     * Group fusion data by genes.
+     * @returns {{[p: string]: StructuralVariant[]}}
+     */
+    @computed
+    get fusionsByGene(): { [hugeGeneSymbol: string]: StructuralVariant[] } {
+        let genes:Array<string> = [];
+        let fusionsByGene:{ [hugeGeneSymbol: string]: StructuralVariant[] } = {};
+        // get string values of genes from input query
+        if (this.genes.result) {
+            // get the genes
+            genes = this.genes.result.map(gene => gene.hugoGeneSymbol);
+            // group fusion data by the genes from input query
+            genes.forEach(gene => {
+                let tmpFusions:Array<any> = [];
+                if (this.fusions.result) {
+                    tmpFusions = tmpFusions.concat(this.fusions.result.filter((fusion:StructuralVariant) => {
+                        return fusion.site1HugoSymbol === gene || fusion.site2HugoSymbol === gene;
+                    }));
+                    fusionsByGene = Object.assign(fusionsByGene, { [gene]: tmpFusions });
+                }
+            });
+        }
+        return fusionsByGene;
+    }
+
+    /**
+     * Assign a FusionMapperStore per genes
+     * @type {MobxPromiseUnionTypeWithDefault<{[p: string]: FusionMapperStore}>}
+     */
+    readonly fusionMapperStores = remoteData<{[hugoGeneSymbol: string]: ResultViewFusionMapperStore}>({
+        invoke: () => {
+            if (this.genes.result && this.fusionsByGene) {
+                return Promise.resolve(
+                    this.genes.result.reduce((map: { [hugoGeneSymbol: string]: ResultViewFusionMapperStore }, gene: Gene) => {
+                        map[gene.hugoGeneSymbol] = new ResultViewFusionMapperStore(
+                            gene,
+                            this.studyIdToStudy,
+                            this.molecularProfileIdToMolecularProfile,
+                            this.samples,
+                            this.fusionsByGene[gene.hugoGeneSymbol]
+                        );
+                        return map;
+                    }, {})
+                );
+            } else {
+                return Promise.resolve({});
+            }
+        }
+    }, {});
+
+    /**
+     * Get fusion mapper store by gene
+     * @param {string} hugoGeneSymbol
+     * @returns {ResultViewFusionMapperStore}
+     */
+    public getFusionMapperStore(hugoGeneSymbol: string): ResultViewFusionMapperStore | undefined {
```

Comment:
this function hides the fact that stores may not exist until genes are loaded.   might be clearer just to reference the fusionMapperStores.result directly in code because developer will then understand they need to handle the loading status.  i don't think this function does enough to warrant the potential confusion it could cause.  (this is common issue in our code base)

Reply:
I agree but also see that it's confusing from @rnugraha POV because this is how we do the mutation tab (from earlier in our work when we didn't have the idioms nailed down)